### PR TITLE
GH47: refactor CDP4Dal.Utils class and split into multiple classes; fixes #47

### LINKED ISSUE #47: refactor CDP4Dal.Utils class and split into multiple classes
### Prerequisites

- [x] I have written a descriptive issue title
- [x] I have verified that I am running the latest version of the CDP4-SDK
- [x] I have searched [open](https://github.com/RHEAGROUP/CDP4-SDK-Community-Edition/issues) and [closed](https://github.com/RHEAGROUP/CDP4-SDK-Community-Edition/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### Description
The `Utils` class in the CDP4Dal namespace is becoming a catch-all for multiple static helper functions that are sometimes implemented as an extension method. This class needs to be refactored into dedicated classes for a specific purpose. I propose the following:

  - `OperationKindExtensions` in the `CDP4Dal.Operations` namespace
  - `UriExtensions` in the `CDP4Dal` namespace
  - `StringExtensions` in the `CDP4Dal` namespace

The `AssertNotNull` can remain in the Utils class

### Steps to Reproduce
NA

### System Configuration

- CDP4 version:
  - [ ] CDP4Common:         
  - [ ] CDP4JsonSerializer: 
  - [x] CDP4Dal:            5.0.0
  - [ ] CDP4JsonFileDal:    
  - [ ] CDP4ServicesDal:    
  - [ ] CDP4WspDal:         
  - [ ] Other:              
- Environment (Operating system, version and so on):
- .NET Framework version:
- Additional information:

### PR DESCRIPTION
**Pull request by nathanatrhea**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/RHEAGROUP/CDP4-SDK-Community-Edition/pulls) open
- [x] I have verified that I am following the CDP4-SDK [code style guidelines](https://raw.githubusercontent.com/RHEAGROUP/CDP4-SDK-Community-Edition/master/.github/CONTRIBUTING.md)
- [x] I have provided test coverage for my change (where applicable)

### Description
- OperationKindExtensions created in CDP4Dal.Operations namespace
- UriExtensions created in the CDP4Dal namespace
- StringExtensions created in the CDP4Dal namespace
- AssertNotNull remains in the Utils Class
- Updated related references

### CODE REVIEW

File: CDP4Dal/Operations/OperationKindExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,71 @@
+﻿#region Copyright
+// --------------------------------------------------------------------------------------------------------------------
+// <copyright file="Utils.cs" company="RHEA System S.A.">
```
Comment (lxatrhea) [inline CDP4Dal/Operations/OperationKindExtensions.cs]: Name doesn't match the class name

File: CDP4Dal/Operations/OperationKindExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,71 @@
+﻿#region Copyright
+// --------------------------------------------------------------------------------------------------------------------
+// <copyright file="Utils.cs" company="RHEA System S.A.">
+//    Copyright (c) 2015-2019 RHEA System S.A.
```
Comment (lxatrhea) [inline CDP4Dal/Operations/OperationKindExtensions.cs]: Please change 2019 to 2020

File: CDP4Dal/Operations/OperationKindExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,71 @@
+﻿#region Copyright
+// --------------------------------------------------------------------------------------------------------------------
+// <copyright file="Utils.cs" company="RHEA System S.A.">
+//    Copyright (c) 2015-2019 RHEA System S.A.
+//
+//    Author: Sam Gerené, Merlin Bieze, Alex Vorobiev, Naron Phou
```
Comment (lxatrhea) [inline CDP4Dal/Operations/OperationKindExtensions.cs]: Feel free to add your own name to all the files you change

File: CDP4Dal/StringExtensions.cs [hunk 4]
```diff
@@ -0,0 +1,78 @@
+﻿#region Copyright
+// --------------------------------------------------------------------------------------------------------------------
+// <copyright file="Utils.cs" company="RHEA System S.A.">
```
Comment (lxatrhea) [inline CDP4Dal/StringExtensions.cs]: Name doesn't match the class name

File: CDP4Dal/StringExtensions.cs [hunk 5]
```diff
@@ -0,0 +1,78 @@
+﻿#region Copyright
+// --------------------------------------------------------------------------------------------------------------------
+// <copyright file="Utils.cs" company="RHEA System S.A.">
+//    Copyright (c) 2015-2019 RHEA System S.A.
```
Comment (lxatrhea) [inline CDP4Dal/StringExtensions.cs]: Please change 2019 to 2020

File: CDP4Dal/StringExtensions.cs [hunk 6]
```diff
@@ -0,0 +1,78 @@
+﻿#region Copyright
+// --------------------------------------------------------------------------------------------------------------------
+// <copyright file="Utils.cs" company="RHEA System S.A.">
+//    Copyright (c) 2015-2019 RHEA System S.A.
+//
+//    Author: Sam Gerené, Merlin Bieze, Alex Vorobiev, Naron Phou
```
Comment (lxatrhea) [inline CDP4Dal/StringExtensions.cs]: Feel free to add your own name to all the files you change

File: CDP4Dal/Operations/OperationKindExtensions.cs [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+//    but WITHOUT ANY WARRANTY; without even the implied warranty of
+//    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+//    Lesser General Public License for more details.
+//
+//    You should have received a copy of the GNU Lesser General Public License
+//    along with this program; if not, write to the Free Software Foundation,
+//    Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+// </copyright>
+// --------------------------------------------------------------------------------------------------------------------
+#endregion
```
Comment (lxatrhea) [inline CDP4Dal/Operations/OperationKindExtensions.cs]: #region and #endregion tags can be removed.
They were added a often at project start (2015) but were at a later stated as deprecated.

File: CDP4Dal/UriExtensions.cs [hunk 8]
```diff
@@ -0,0 +1,70 @@
+﻿#region Copyright
+// --------------------------------------------------------------------------------------------------------------------
+// <copyright file="Utils.cs" company="RHEA System S.A.">
```
Comment (lxatrhea) [inline CDP4Dal/UriExtensions.cs]: Name doesn't match the class name

File: CDP4Dal/UriExtensions.cs [hunk 9]
```diff
@@ -0,0 +1,70 @@
+﻿#region Copyright
+// --------------------------------------------------------------------------------------------------------------------
+// <copyright file="Utils.cs" company="RHEA System S.A.">
+//    Copyright (c) 2015-2019 RHEA System S.A.
```
Comment (lxatrhea) [inline CDP4Dal/UriExtensions.cs]: Please change 2019 to 2020

File: CDP4Dal/UriExtensions.cs [hunk 10]
```diff
@@ -0,0 +1,70 @@
+﻿#region Copyright
+// --------------------------------------------------------------------------------------------------------------------
+// <copyright file="Utils.cs" company="RHEA System S.A.">
+//    Copyright (c) 2015-2019 RHEA System S.A.
+//
+//    Author: Sam Gerené, Merlin Bieze, Alex Vorobiev, Naron Phou
```
Comment (lxatrhea) [inline CDP4Dal/UriExtensions.cs]: Feel free to add your own name to all the files you change

**Comment by alexatrhea** [comment]
AppVeyor:

    4525 Warning(s)
    **36 Error(s)**
Time Elapsed 00:09:17.51

**Pull request merged**